lib/src/wolt_modal_sheet.dart

```diff
@@ -424,4 +533,231 @@ class _WoltModalSheetState extends State<WoltModalSheet> {
+    return addOrReplacePages([page]);
+  }
+
+  /// Adds one or more new pages to the modal sheet stack and displays the first of the added
+  /// page list.
+  ///
+  /// This method pushes a list of [SliverWoltModalSheetPage] onto the navigation stack,
+  /// adding them to the end. It then updates the displayed page to be the first page in the newly added list,
+  /// allowing for a sequence of pages to be added and immediately interacted with.
+  ///
+  /// Parameters:
+  /// - [pages]: The list of [SliverWoltModalSheetPage] to add to the navigation stack.
+  ///
+  /// Returns:
+  /// None.
+  void pushPages(List<SliverWoltModalSheetPage> pages) {
+    setState(() {
+      _pages.addAll(pages);
+      if (pages.isNotEmpty) {
+        _selectedPageIndex =
+            _pages.length - pages.length; // Set to the first of the newly added pages.
+      }
+    });
+  }
+
+  /// Adds a new page to the modal sheet stack and displays it.
+  ///
+  /// This method pushes a new [SliverWoltModalSheetPage] onto the navigation stack,
+  /// making it the visible page. If the modal sheet is currently showing another page,
+  /// that page will be transitioned by the new page.
+  ///
+  /// Parameters:
+  /// - [page]: The [SliverWoltModalSheetPage] to add to the navigation stack.
+  ///
+  /// Returns:
+  /// None.
+  void pushPage(SliverWoltModalSheetPage page) {
+    pushPages([page]); // Use the pushPages method to handle the single page push.
+  }
+
+  /// Removes the top page from the modal sheet stack.
+  ///
+  /// This method pops the top [SliverWoltModalSheetPage] from the navigation stack.
+  /// If there's only one page left in the stack, this operation will not remove it,
+  /// ensuring that there's always at least one page displayed.
+  ///
+  /// Returns:
+  /// - [bool]: Returns `true` if a page was successfully removed; `false` if the stack
+  ///   had only one page left.
+  bool popPage() {
+    if (_pages.length > 1) {
+      setState(() {
+        _pages.removeLast();
+        _selectedPageIndex = _pages.length - 1;
+      });
+      return true;
+    }
+    return false;
+  }
+
+  /// Replaces a specific page identified by [id] with a new page [page].
+  ///
+  /// Searches for a page with the given [id] and replaces it with the new [page]. If no page with
+  /// the specified [id] is found, an error is thrown.
+  ///
+  /// Parameters:
+  /// - [id]: The identifier of the page to replace.
+  /// - [page]: The new [SliverWoltModalSheetPage] to insert in place of the old one.
+  ///
+  /// Returns:
+  /// - [bool]: Returns `true` if replacing  page was successful, `false` if no page with the
+  /// specified [id] is found.
+  bool replacePage(Object id, SliverWoltModalSheetPage page) {
+    bool found = false;
+    final List<SliverWoltModalSheetPage> newPages = _pages.map((p) {
+      if (p.id == id) {
+        found = true;
+        return page;
+      }
+      return p;
+    }).toList();
+    if (!found) return false;
```

Comment:
```suggestion
    final List<SliverWoltModalSheetPage> newPages = _pages.where((SliverWoltModalSheetPage sheetPage) {
      return sheetPage.id == id;
    }).toList();
    if (newPages.isEmpty) {
      return false;
    }
```

This could where and avoid using inline veriables. I haven't tested the code so please verify it works.